docs/assert/timeout.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: default
+title: timeout
+description: Sets the length of time to wait for async operations before failing the test.
+categories:
+  - assert
+  - async
+---
+
+## `timeout( duration )`
+
+Sets the length of time to wait for async operations before failing the test.
+
+| name | description |
+|------|-------------|
+| `duration` (Number) | The length of time, in milliseconds, to wait for async operations. |
+
+### Description
+
+`assert.timeout()` sets the length of time, in milliseconds, to wait for async operations in the current test. This is equivalent to setting `config.testTimeout` on a per-test basis. The timeout length only applies when performing async operations.
+
+If `0` is passed, then the test will be assumed to be completely synchronous. If a non-numeric value is passed as an argument, the function will throw an error.
```

Comment:
To summarize (to ensure I'm understanding everything):
* `setTimeout(, 0)` doesn't guarantee synchronicity (agreed)
* However, we _do want_ to guarantee synchronicity when using a timeout of `0`
* Therefore, we should special case `assert.timeout(0)`
* And, add tests handling edge cases that likely fail with the current implementation
Does that seem correct?